okhttp/src/main/java/okhttp3/internal/connection/RouteDatabase.kt

```diff
@@ -13,33 +13,32 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package okhttp3.internal.connection;
+package okhttp3.internal.connection
 
-import java.util.LinkedHashSet;
-import java.util.Set;
-import okhttp3.Route;
+import okhttp3.Route
+import java.util.LinkedHashSet
 
 /**
  * A blacklist of failed routes to avoid when creating a new connection to a target address. This is
  * used so that OkHttp can learn from its mistakes: if there was a failure attempting to connect to
  * a specific IP address or proxy server, that failure is remembered and alternate routes are
  * preferred.
  */
-final class RouteDatabase {
-  private final Set<Route> failedRoutes = new LinkedHashSet<>();
+class RouteDatabase {
+  private val failedRoutes = LinkedHashSet<Route>()
```

Comment:
Specify the type for `failedRoutes` explicitly to prevent its type determined as `LinkedHashSet<Route>`.
```kotlin
private val failedRoutes: Set<Route> = ...
```

Reply:
`add` and `remove` aren't available in `Set`. I've used `HashSet` instead:
`private val failedRoutes: HashSet<Route> = linkedSetOf()`

Reply:
You can use `mutableSetOf()` instead to use `add` or `remove`.

Reply:
use MutableSet

Reply:
This method is more the kotlin-way.
```kotlin
private val failedRoutes: MutableSet<Route> = mutableSetOf()
```

Reply:
Yep, i missed that it was linkedHashSet by default.  TIL

Reply:
In short, it can be like:
```kotlin
private val failedRoutes = mutableSetOf<Route>()
```
The type of failedRoutes is still `MutableSet<Route>`.

Reply:
Thanks. Updated and rebased.